src/transformers/trainer_pt_utils.py

```diff
@@ -1395,3 +1395,211 @@ def get_lr(self):
 
     def _get_closed_form_lr(self):
         return self.base_lrs
+
+
+@dataclass
+class FSDPConfig:
```

Comment:
## Code change comments


New dataclass similar to `AcceleratorConfig` is added